fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -731,6 +736,57 @@ where
     }
 }
 
+impl<M, S, Tx, Ecal> Interpreter<M, S, Tx, Ecal>
+where
+    S: InterpreterStorage,
+{
+    fn blob_inner(
+        blob: &mut Blob,
+        storage: &mut S,
+        initial_balances: InitialBalances,
+        gas_costs: &GasCosts,
+        fee_params: &FeeParameters,
+        base_asset_id: &AssetId,
+        gas_price: Word,
+    ) -> Result<(), InterpreterError<S::DataError>> {
+        let blob_data = blob
+            .witnesses()
+            .get(*blob.bytecode_witness_index() as usize)
+            .ok_or(InterpreterError::Bug(Bug::new(
+                // It shouldn't be possible since `Checked<Blob>` guarantees
+                // the existence of the witness.
+                BugVariant::WitnessIndexOutOfBounds,
+            )))?;
+
+        let blob_id = blob.blob_id();
+
+        debug_assert_eq!(
+            BlobId::compute(blob_data.as_ref()),
+            *blob_id,
+            "Tx has invalid BlobId",
+        );
+
+        storage
```

Comment:
We need to return an error, if the same blob id already exists.

Reply:
Done in d38fc654